src/test/TestGhoStewardsForkRemote.t.sol

```diff
@@ -0,0 +1,338 @@
+      isEnabled: outboundConfig.isEnabled,
+      capacity: inboundConfig.capacity,
+      rate: inboundConfig.rate
+    });
+
+    vm.expectRevert('Only callable by owner');
+    vm.prank(RISK_COUNCIL);
+    GHO_CCIP_STEWARD.updateRateLimit(
+      remoteChainSelector,
+      newOutboundConfig.isEnabled,
+      newOutboundConfig.capacity,
+      newOutboundConfig.rate,
+      newInboundConfig.isEnabled,
+      newInboundConfig.capacity,
+      newInboundConfig.rate
+    );
+  }
+
+  function testGhoCcipStewardUpdateRateLimitAfterPoolUpgrade() public {
+    MockUpgradeableBurnMintTokenPool tokenPoolImpl = new MockUpgradeableBurnMintTokenPool(
+      address(GHO_TOKEN),
+      address(ARM_PROXY),
+      false,
+      false
+    );
+
+    vm.prank(PROXY_ADMIN);
+    TransparentUpgradeableProxy(payable(address(GHO_TOKEN_POOL))).upgradeTo(address(tokenPoolImpl));
+
+    vm.prank(ACL_ADMIN);
+    IUpgradeableLockReleaseTokenPool(GHO_TOKEN_POOL).setRateLimitAdmin(address(GHO_CCIP_STEWARD));
+
+    RateLimiter.TokenBucket memory mockConfig = RateLimiter.TokenBucket({
+      rate: 50,
+      capacity: 50,
+      tokens: 1,
+      lastUpdated: 1,
+      isEnabled: true
+    });
+
+    // Mocking response due to rate limit currently being 0
+    vm.mockCall(
+      GHO_TOKEN_POOL,
+      abi.encodeWithSelector(
+        IUpgradeableLockReleaseTokenPool(GHO_TOKEN_POOL)
+          .getCurrentOutboundRateLimiterState
+          .selector,
+        remoteChainSelector
+      ),
+      abi.encode(mockConfig)
+    );
+    vm.mockCall(
+      GHO_TOKEN_POOL,
+      abi.encodeWithSelector(
+        IUpgradeableLockReleaseTokenPool(GHO_TOKEN_POOL).getCurrentInboundRateLimiterState.selector,
+        remoteChainSelector
+      ),
+      abi.encode(mockConfig)
+    );
+
+    RateLimiter.TokenBucket memory outboundConfig = IUpgradeableLockReleaseTokenPool(GHO_TOKEN_POOL)
+      .getCurrentOutboundRateLimiterState(remoteChainSelector);
+    RateLimiter.TokenBucket memory inboundConfig = IUpgradeableLockReleaseTokenPool(GHO_TOKEN_POOL)
+      .getCurrentInboundRateLimiterState(remoteChainSelector);
+
+    RateLimiter.Config memory newOutboundConfig = RateLimiter.Config({
+      isEnabled: outboundConfig.isEnabled,
+      capacity: outboundConfig.capacity + 1,
+      rate: outboundConfig.rate
+    });
+
+    RateLimiter.Config memory newInboundConfig = RateLimiter.Config({
+      isEnabled: outboundConfig.isEnabled,
+      capacity: inboundConfig.capacity + 1,
+      rate: inboundConfig.rate
+    });
+
+    vm.expectEmit(false, false, false, true);
+    emit ChainConfigured(remoteChainSelector, newOutboundConfig, newInboundConfig);
+    vm.prank(RISK_COUNCIL);
+    GHO_CCIP_STEWARD.updateRateLimit(
+      remoteChainSelector,
+      newOutboundConfig.isEnabled,
+      newOutboundConfig.capacity,
+      newOutboundConfig.rate,
+      newInboundConfig.isEnabled,
+      newInboundConfig.capacity,
+      newInboundConfig.rate
+    );
+  }
```

Comment:
assert new rate config is applied

Reply:
I can do that, but I have the event emission checking easily in one line, which is the pattern I'd been using for these tests

Reply:
ah, it's fine then